flink-connector-prometheus/src/main/java/org/apache/flink/connector/prometheus/sink/HttpResponseCallback.java

```diff
@@ -156,35 +170,35 @@ public void completed(SimpleHttpResponse response) {
         }
 
         // Unexpected/unhandled response outcome
-        throw new PrometheusSinkWriteException(
-                "Unexpected status code returned from the remote-write endpoint",
-                statusCode,
-                reasonPhrase,
-                timeSeriesCount,
-                sampleCount,
-                responseBody);
+        logErrorAndThrow(
+                new PrometheusSinkWriteException(
+                        "Unexpected status code returned from the remote-write endpoint",
+                        statusCode,
+                        reasonPhrase,
+                        timeSeriesCount,
+                        sampleCount,
+                        responseBody));
     }
 
     @Override
     public void failed(Exception ex) {
         // General I/O failure reported by http client
-        // Depending on the configured behavior, throw an exception or log and discard
-        if (errorHandlingBehaviorConfig.getOnHttpClientIOFail()
-                == PrometheusSinkConfiguration.OnErrorBehavior.FAIL) {
-            throw new PrometheusSinkWriteException("Http client failure", ex);
-        } else {
-            LOG.warn(
-                    "Exception executing the remote-write (discarded {} time-series containing {} samples)",
-                    timeSeriesCount,
-                    sampleCount,
-                    ex);
-            metricsCallback.onFailedWriteRequestForHttpClientIoFail(sampleCount);
-        }
+        // Always fail
+        throw new PrometheusSinkWriteException("Http client failure", ex);
```

Comment:
Is there a reason we don't want to add the sampleCount etc here?

Reply:
There is no context when an exception is thrown by the client. We do not know how many samples the request did contain.
Anyhow, it is not important because this always make the operator fail. Failure counters are only useful to measure dropped data on DISCARD_AND_CONTINUE